test/Datadog.Trace.Tests/TracerTests.cs

```diff
@@ -432,19 +431,40 @@ public async Task DisconnectRemoteObjectsAfterCrossDomainCallsOnDispose()
                 AppDomain.Unload(remote);
             }
 
-            await Task.Delay(200);
+            // Ensure that we wait long enough for the lease manager poll to occur.
+            // Even though we reset LifetimeServices.LeaseManagerPollTime to a shorter duration,
+            // the default value is 10 seconds so the first poll may not be affected by our modification
+            cde.Wait(TimeSpan.FromSeconds(30));
```

Comment:
You should probably have an assertion here. If Wait returns false, something wrong is going on 
(though the DisconnectCount assertion will then likely fail, so that's OK-ish)

Reply:
@kevingosse: How's this? 45bd4d6e

Reply:
Great, thanks!